.github/workflows/run-ci-for-external-forks.yml

```diff
@@ -0,0 +1,39 @@
+# This workflow cleans up any leftover projects created by e2e runs.
+
+name: Run CI on behalf of External Forks
+on:
+  workflow_dispatch:
+    inputs:
+      pr-number:
+        description: "The PR number to run CI on behalf of"
+        required: true
+      reviewed:
+        description: "Confirm that the PR has been reviewed for use/leakage of secrets"
+        type: boolean
+        required: true
+jobs:
+  create-draft-pr:
+    if: ${{ inputs.reviewed == true }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check user for team affiliation
+        uses: tspascoal/get-user-teams-membership@v2
+        id: teamAffiliation
+        with:
+          GITHUB_TOKEN: ${{ secrets.GH_ACCESS_TOKEN }}
+          username: ${{ github.actor }}
+          team: wrangler
+      - name: Stop workflow if user is no member
+        if: ${{ steps.teamAffiliation.outputs.isTeamMember == false }}
+        run: |
+          echo "You have must be on the "wrangler" team to trigger this job."
+          exit 1
+
+      - name: "Checkout PR"
+        run: gh pr checkout ${{ inputs.pr-number }}
+
+      - name: "Create Draft PR"
+        run: git checkout -b run-ci-on-hehalf-of-${{ inputs.pr-number }}
```

Comment:
Should we run a `branch -D` on this branch first? Unlikely conflict, but possible.

Reply:
Yeah I was considering this needs to be repeatable. I was thinking a force push or something once I confirmed this all worked but `branch -D` would also work – do we need to delete the remote branch too? and close the existing PR?

Reply:
If that's possible then it seems like a good idea, to avoid needing any potential manual cleanup.